### PR TITLE
removed m dir from default configuration

### LINKED ISSUE #1302: We should rethink about a few default configuration
This made sense as we were writting to the disk, it partly does not anymore.
```
    // Output directories for storing js and css resources                                                                                                                                                    
    dirs: {
      style: 's',
      media: 'm',
      javascript: 'j',
      styleModules: 'css_modules',
      jsModules: 'js_modules',
    },
```

### PR DESCRIPTION
**Pull request by MananJethwani**
@kelson42 this resolves #1302 
do let me know if something else needed to be removed I assume js/jsModules and style/styleModulle needs to be merged as well

### CODE REVIEW

File: src/util/misc.ts [hunk 1]
```diff
@@ -248,7 +248,8 @@ export function getMediaBase(url: string, escape: boolean, dir: string = config.
       filename = crypto.createHash('md5').update(decodedUrl).digest('hex') + path.extname((new URL(url)).pathname);
   }
 
-  return `${dir}/${ escape ? encodeURIComponent(filename) : filename }`;
+  // change here
```
Comment (kelson42) [inline src/util/misc.ts:251]: This comment has to be removed.

**Review by kelson42 — CHANGES_REQUESTED** [review]
Looks good, we should evalutes the other paths in the configuration. The thing to take care is to evaluate if we are not creating potential collisions in naming by doing so.

File: src/util/misc.ts (hunk 1 above)
Comment (MananJethwani) [reply src/util/misc.ts:251]: oh ok...will remove it

**Review by kelson42 — CHANGES_REQUESTED** [review]
`genHeaderScript` or `genHeaderScript` should have a default value as third argument. That will avoid you to change all the calls. I suspect that in place of a boolean, we could have advantageously the subdirectory value itself.

File: test/unit/util.test.ts [hunk 2]
```diff
@@ -59,12 +59,12 @@ test('Get full URL', async(t) => {
...
-    t.equal(getFullUrl('./-/j/js_modules/jsConfigVars.js', new URL('https://bm.wikipedia.org/')),
-            'https://bm.wikipedia.org/-/j/js_modules/jsConfigVars.js',
+    t.equal(getFullUrl('./-/sources/jsConfigVars.js', new URL('https://bm.wikipedia.org/')),
+            'https://bm.wikipedia.org/-/sources/jsConfigVars.js',
             'Full Url for relative path with skipping one file');
 
-    t.equal(getFullUrl('../-/j/js_modules/jsConfigVars.js', 'https://bm.wikipedia.org/'),
-            'https://bm.wikipedia.org/-/j/js_modules/jsConfigVars.js',
+    t.equal(getFullUrl('../-/sources/jsConfigVars.js', 'https://bm.wikipedia.org/'),
+            'https://bm.wikipedia.org/-/sources/jsConfigVars.js',
```
Comment (kelson42) [inline test/unit/util.test.ts:67]: Where is this « sources » coming from?

File: src/util/misc.ts [hunk 3]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
...
+  return `<link href="${upStr}${resourceNamespace}/${cssPath(css, subDirectory)}" rel="stylesheet" type="text/css" class="${classList}" />`;
 }
-export function genHeaderScript(config: Config, js: string, articleId: string, classList = '') {
+export function genHeaderScript(config: Config, js: string, articleId: string, addSubDirectory = '', classList = '') {
   const resourceNamespace = '-';
   const slashesInUrl = articleId.split('/').length - 1;
   const upStr = '../'.repeat(slashesInUrl + 1);
   const path = (isNodeModule(js)) ? normalizeModule(js) : js;
-  return `<script src="${upStr}${resourceNamespace}/${jsPath(config, path)}" class="${classList}"></script>`;
+  return `<script src="${upStr}${resourceNamespace}/${jsPath(path, addSubDirectory)}" class="${classList}"></script>`;
```
Comment (kelson42) [inline src/util/misc.ts:194]: it ahiuld be subDirectory, not addSubDirectory.

**Review by kelson42 — CHANGES_REQUESTED** [review]

File: test/unit/util.test.ts (hunk 2 above)
Comment (MananJethwani) [reply test/unit/util.test.ts:67]: might have been caused after rebasing extremely sorry

**Comment by MananJethwani** [comment]
@kelson42 sorry for those errors.......might have been caused while I rebased
please review when free

File: src/util/misc.ts [hunk 4]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
   ]);
 }
 
-export function cssPath({ output: { dirs } }: Config, css: string) {
-  return [dirs.style, `${dirs.styleModules}/${css.replace(/(\.css)?$/, '')}.css`].join('/');
+export function cssPath(css: string, subDirectory = '') {
```
Comment (kelson42) [inline src/util/misc.ts:176]: We should maybe add type checking here for `subDirectory`?

File: src/util/misc.ts [hunk 5]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
...
+  return `${subDirectory ? `${subDirectory}/` : ''}${css.replace(/(\.css)?$/, '')}.css`;
```
Comment (kelson42) [inline src/util/misc.ts:177]: Do we still need to check `subDirectory` now that we have a default value?

File: src/util/misc.ts [hunk 6]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
...
 }
-export function jsPath({ output: { dirs } }: Config, js: string) {
+export function jsPath(js: string, subDirectory = '') {
```
Comment (kelson42) [inline src/util/misc.ts:179]: `string` typechecking?

File: src/util/misc.ts [hunk 7]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
...
   const path = (isNodeModule(js)) ? normalizeModule(js) : js;
-  return [dirs.javascript, `${dirs.jsModules}/${path.replace(/(\.js)?$/, '')}.js`].join('/');
+  return `${subDirectory ? `${config.output.dirs.mediawiki}/` : ''}${path.replace(/(\.js)?$/, '')}.js`;
```
Comment (kelson42) [inline src/util/misc.ts:181]: Do we still need to check `subDirectory`?

File: src/util/misc.ts [hunk 8]
```diff
@@ -173,25 +173,25 @@ export function saveStaticFiles(config: Config, zimCreator: ZimCreator) {
...
-export function genHeaderCSSLink(config: Config, css: string, articleId: string, classList = '') {
+export function genHeaderCSSLink(config: Config, css: string, articleId: string, subDirectory = '', classList = '') {
   const resourceNamespace = '-';
   const slashesInUrl = articleId.split('/').length - 1;
   const upStr = '../'.repeat(slashesInUrl + 1);
-  return `<link href="${upStr}${resourceNamespace}/${cssPath(config, css)}" rel="stylesheet" type="text/css" class="${classList}" />`;
+  return `<link href="${upStr}${resourceNamespace}/${cssPath(css, subDirectory)}" rel="stylesheet" type="text/css" class="${classList}" />`;
 }
-export function genHeaderScript(config: Config, js: string, articleId: string, classList = '') {
+export function genHeaderScript(config: Config, js: string, articleId: string, subDirectory = '', classList = '') {
```
Comment (kelson42) [inline src/util/misc.ts:189]: typechecking? For `classlist` as well?

**Review by kelson42 — CHANGES_REQUESTED** [review]

File: src/util/misc.ts (hunk 4 above)
Comment (MananJethwani) [reply src/util/misc.ts:176]: ok will add that as well

File: src/util/misc.ts (hunk 5 above)
Comment (MananJethwani) [reply src/util/misc.ts:177]: it's needed as we are passing just the subdirectory name we need to add `/` between them that's why I used this method

File: src/util/misc.ts (hunk 6 above)
Comment (MananJethwani) [reply src/util/misc.ts:179]: sure

File: src/util/misc.ts (hunk 8 above)
Comment (MananJethwani) [reply src/util/misc.ts:189]: sure....but one thing I wanted to point out I didn't see class list being passed anywhere while calling these function will we need them at some point

File: src/util/misc.ts (hunk 7 above)
Comment (MananJethwani) [reply src/util/misc.ts:181]: same as above for `/`

File: src/util/misc.ts (hunk 8 above)
Comment (kelson42) [reply src/util/misc.ts:189]: then please remove it.

**Review by kelson42 — APPROVED** [review]

**Pull request merged**